src/audiomixerboard.cpp

```diff
@@ -908,6 +958,20 @@ void CAudioMixerBoard::UpdateGainValue ( const int    iChannelIdx,
     emit ChangeChanGain ( iChannelIdx, dValue );
 }
 
+void CAudioMixerBoard::OnPanValueChanged ( const double dValue )
+{
+
+    // get channel index from sender according t, this SHOULD coincide with the chanID.
+    CChannelFader * channelFader = qobject_cast<CChannelFader * >(sender());
```

Comment:
I want to have consistency in the code. The pan can be treated exactly like the fader value of that channel. I do not want to have a different type of implementation for the pan.

Reply:
I understand. The reason to try out this solution was since creating 50 more signals and slots (like fader) seemed not so elegant - as it was when I started to work on this code. Now you have created a conditional  way to reduce it for Qt5 (using connectFaderSignalsToMixerBoardSlots and the slotID) but maybe there is still a way to keep it simpler for Qt4, too? 

I am not sure, is it so or can it be used so that the channelID of a fader/pan can be the same as index of the according CChannelFader object?? Then perhaps just we could add a field iChanID to CChannelFader, set it on creation and make in the same time the connections? Something like this in CAudioMixerBoard::CAudioMixerBoard:

for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
    {
        vecpChanFader[i] = new CChannelFader ( this, pMainLayout );
        vecpChanFader[i]->Hide();
       vecpChanFader[i]->chanID = i;
       QObject::connect ( vecpChanFader[i],  SIGNAL ( panValueChanged ( double ) ),
                           this, SLOT ( OnPanValueChanged ( double ) ) ); 
      QObject::connect ( vecpChanFader[i],  SIGNAL ( gainValueChanged ( double ) ),
                           this, SLOT ( OnGainValueChanged ( double ) ) );
    }

and the slot gets the iChanID as:

CChannelFader * channelFader = qobject_cast<CChannelFader * >(sender());
int iChanID = channelFade->iChanID;
emit ChangeChanPan ( iChanID, dValue );

(sorry this code is just as an example, not correct).

Reply:
I know that it is not nice to have 50 lines of code for each connection. All this code will be cleaned up as soon as I drop the support for Qt4. My plan is to drop support of Qt4 as soon as Qt6 is available for the Open Source community (which will most likely be in about a year). Until then I want to keep the code consistent. I do not want to have the Pan differently implemented than all the others. So please do it exactly like the faders are implemented.